src/awscdk/github.ts

```diff
@@ -230,9 +215,24 @@ export class GithubCDKPipeline extends CDKPipeline {
    * @param stage - The deployment stage to create.
    */
   public createDeployment(stage: DeploymentStage): void {
-    const preInstallSteps = (this.options.preInstallSteps ?? []).map(s => s.toGithub());
 
     if (stage.manualApproval === true) {
+      const steps = [
+        new AwsAssumeRoleStep(this.project, {
+          roleArn: this.options.iamRoleArns?.deployment?.[stage.name] ?? this.options.iamRoleArns?.default!,
+          region: stage.env.region,
+        }),
+        ...this.options.preInstallSteps ?? [],
+        new SimpleCommandStep(this.project, this.renderInstallCommands()),
+        new SimpleCommandStep(this.project, this.renderInstallPackageCommands(`${this.options.pkgNamespace}/${this.app.name}@\${{github.event.inputs.version}}`)),
+        new SimpleCommandStep(this.project, [`mv ./node_modules/${this.options.pkgNamespace}/${this.app.name} ${this.app.cdkConfig.cdkout}`]),
```

Comment:
hard coded for a node/typscript project?

Reply:
The whole CDK pipeline is for TypeScript projects.

Reply:
In this case we are using a NPM registry as temporary artifact space for the cloud assembly. SO this would also work for other projects as it only needs to match the artifact upload step

Reply:
Got it .